src/publishPackages/publish.proj

```diff
@@ -53,33 +55,84 @@
+  <Target Name="PreparePublishPackages" DependsOnTargets="PreparePublish">
     <PropertyGroup>
-      <AssetManifestFilePath>$(ManifestDir)reference-packages-nupkg-manifest.xml</AssetManifestFilePath>
+      <AssetManifestFilename>reference-packages-nupkg-manifest.xml</AssetManifestFilename>
+      <AssetManifestFilePath>$(ManifestDir)$(AssetManifestFileName)</AssetManifestFilePath>
+      <!-- Create a temporary directory to store the generated asset manifest by the task -->
+      <TempWorkingDirectory>$(ArtifactsDir)AssetsTmpDir\$([System.Guid]::NewGuid())/</TempWorkingDirectory>
     </PropertyGroup>
     <ItemGroup>
       <PackagesToPublish Include="$(ArtifactsPkgDir)**/*.nupkg" />
     </ItemGroup>
+    <MakeDir Directories="$(TempWorkingDirectory)"/>
+    <Message Importance="High" Text="[$([System.DateTime]::Now.ToString('HH:mm:ss.ff'))] DB1 -- :: Files to Publish @(PackagesToPublish) " />
+    <PushToAzureDevOpsArtifacts
+      ItemsToPush="@(PackagesToPublish)"
+      ManifestBuildData="Location=$(ExpectedFeedUrl)"
+      ManifestRepoUri="$(ManifestName)"
+      ManifestBranch="$(ManifestBranch)"
+      ManifestBuildId="$(ManifestBuildId)"
+      ManifestCommit="$(ManifestCommit)"
+      IsStableBuild="false"
+      AssetManifestPath="$(AssetManifestFilePath)"
+      AssetsTemporaryDirectory="$(TempWorkingDirectory)" />
+
+    <!-- Copy the generated manifest to the build's artifacts -->
+    <Copy
+      SourceFiles="$(AssetManifestFilePath)"
+      DestinationFolder="$(TempWorkingDirectory)$(AssetManifestFileName)" />
+
+    <Message
+      Text="##vso[artifact.upload containerfolder=AssetManifests;artifactname=AssetManifests]$(TempWorkingDirectory)$(AssetManifestFileName)"
+      Importance="high" />
+
   </Target>
 
-  <Target Name="PreparePublishFiles" DependsOnTargets="PreparePublish;GetTarballFileName" BeforeTargets="PublishFilesToBlobFeed">
+  <Target Name="PreparePublishFiles" DependsOnTargets="PreparePublish;GetTarballFileName">
     <PropertyGroup>
-      <AssetManifestFilePath>$(ManifestDir)reference-packages-tar-manifest.xml</AssetManifestFilePath>
+      <AssetManifestFilename>reference-packages-tar-manifest.xml</AssetManifestFilename>
+      <AssetManifestFilePath>$(ManifestDir)$(AssetManifestFileName)</AssetManifestFilePath>
+      <!-- Create a temporary directory to store the generated asset manifest by the task -->
+      <TempWorkingDirectory>$(ArtifactsDir)AssetsTmpDir/$([System.Guid]::NewGuid())/</TempWorkingDirectory>
     </PropertyGroup>
     <ItemGroup>
       <FilesToPublish Include="$(TarballFileName)" />
     </ItemGroup>
-  </Target>
+    <MakeDir Directories="$(TempWorkingDirectory)"/>
+    <Message Importance="High" Text="[$([System.DateTime]::Now.ToString('HH:mm:ss.ff'))] DB2 -- :: Files to Publish @(FilesToPublish) " />
+    <PushToAzureDevOpsArtifacts
+      ItemsToPush="@(FilesToPublish)"
+      ManifestBuildData="Location=$(ExpectedFeedUrl)"
```

Comment:
The code will work as it is. But it would be better if you could include other build data attributes here. Like what's done here: https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj#L167